component/prometheus/exporter/oracledb/oracledb.go

```diff
@@ -0,0 +1,88 @@
+package oracledb
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/prometheus/exporter"
+	"github.com/grafana/agent/pkg/integrations"
+	"github.com/grafana/agent/pkg/integrations/oracledb_exporter"
+	"github.com/grafana/agent/pkg/river/rivertypes"
+	config_util "github.com/prometheus/common/config"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:    "prometheus.exporter.oracledb",
+		Args:    Arguments{},
+		Exports: exporter.Exports{},
+		Build:   exporter.New(createExporter, "oracledb"),
+	})
+}
+
+func createExporter(opts component.Options, args component.Arguments) (integrations.Integration, error) {
+	a := args.(Arguments)
+	return a.Convert().NewIntegration(opts.Logger)
+}
+
```

Comment:
This is something very new, but flow prometheus exporter components were missing the customized `instance` label required for compatibility with Grafana Cloud integrations. We have introduced them in #3873. Would you mind adding a function similar to https://github.com/grafana/agent/blob/f53714a7df98cb03185202f4caa839e69d8520dd/component/prometheus/exporter/mssql/mssql.go#L31-L42 and passing it like it is done [here](https://github.com/grafana/agent/blob/f53714a7df98cb03185202f4caa839e69d8520dd/component/prometheus/exporter/mssql/mssql.go#L22)?

The logic that the function needs to implement is defined [here](https://github.com/grafana/agent/blob/f53714a7df98cb03185202f4caa839e69d8520dd/pkg/integrations/oracledb_exporter/oracledb_exporter.go#L78-L84).


Here are some example [tests](https://github.com/grafana/agent/blob/f53714a7df98cb03185202f4caa839e69d8520dd/component/prometheus/exporter/mssql/mssql_test.go#L129-L151).

Reply:
No problem. Updated to your suggestions @jcreixell !